src/Persistence/PaYaml/Serialization/PaYamlSerializer.cs

```diff
@@ -115,6 +115,10 @@ private static void WriteTextWriter<TValue>(TextWriter writer, in TValue? value,
 
             return value;
         }
+        catch (YamlException ex) when (ex.InnerException is ArgumentException && ex.InnerException.Message.Contains("An item with the same key has already been added"))
```

Comment:
Instead of adding another catch clause, you can change the existing one to compute the error code based on the where clause.